js/components/braveryPanel.js

```diff
@@ -0,0 +1,240 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const React = require('react')
+const ipc = require('electron').ipcRenderer
+const ImmutableComponent = require('./immutableComponent')
+const Dialog = require('./dialog')
+const SwitchControl = require('./switchControl')
+const windowActions = require('../actions/windowActions')
+const appActions = require('../actions/appActions')
+const urlParse = require('url').parse
+const cx = require('../lib/classSet')
+const messages = require('../constants/messages')
+
+class BraveryPanel extends ImmutableComponent {
+  constructor () {
+    super()
+    this.onToggleSiteSetting = this.onToggleSiteSetting.bind(this)
+    this.onToggleBlockedAds = this.onToggleBlockedAds.bind(this)
+    this.onToggleTPList = this.onToggleTPList.bind(this)
+    this.onToggleHttpseList = this.onToggleHttpseList.bind(this)
+    this.onToggleAdvanced = this.onToggleAdvanced.bind(this)
+    this.onToggleShields = this.onToggleSiteSetting.bind(this, 'shieldsUp')
+    this.onToggleAdControl = this.onToggleSiteSetting.bind(this, 'adControl')
+    this.onToggleSafeBrowsing = this.onToggleSiteSetting.bind(this, 'safeBrowsing')
+    this.onToggleNoScript = this.onToggleSiteSetting.bind(this, 'noScript')
+    this.onToggleCookieControl = this.onToggleSiteSetting.bind(this, 'cookieControl')
+    this.onToggleHTTPSE = this.onToggleSiteSetting.bind(this, 'httpsEverywhere')
+    this.onReload = this.onReload.bind(this)
+  }
+  get isExtendedValidation () {
+    return this.props.frameProps.getIn(['security', 'isExtendedValidation'])
+  }
+  get isSecure () {
+    return this.props.frameProps.getIn(['security', 'isSecure'])
+  }
+  get isBlockingTrackedContent () {
+    return this.blockedByTrackingList && this.blockedByTrackingList.size > 0
+  }
+  get isMixedContent () {
+    return this.props.frameProps.getIn(['security', 'isMixedContent'])
+  }
+  get blockedByTrackingList () {
+    return this.props.frameProps.getIn(['trackingProtection', 'blocked'])
+  }
+  get isTPListShown () {
+    return this.props.braveryPanelDetail.get('expandTrackingProtection')
+  }
+  get isAdvancedExpanded () {
+    return this.props.braveryPanelDetail.get('advancedControls') !== false
+  }
+  get blockedAds () {
+    return this.props.frameProps.getIn(['adblock', 'blocked'])
+  }
+  get isBlockingAds () {
+    return this.blockedAds && this.blockedAds.size > 0
+  }
+  get isBlockedAdsShown () {
+    return this.props.braveryPanelDetail.get('expandAdblock')
+  }
+  get isHttpseShown () {
+    return this.props.braveryPanelDetail.get('expandHttpse')
+  }
+  get redirectedResources () {
+    return this.props.frameProps.get('httpsEverywhere')
+  }
+  get isRedirectingResources () {
+    return this.redirectedResources && this.redirectedResources.size > 0
+  }
+  get origin () {
+    const parsedUrl = urlParse(this.props.frameProps.get('location'))
+    return `${parsedUrl.protocol}//${parsedUrl.host}`
```

Comment:
that should be moved to urlUtil because siteUtil is only for dealing with site state. I'll use it for now though.